develop/CICD/GitHub/GitHubReusableWorkflows/GitHub_Reusable_Workflows_NuGet_Solution_Master_Workflow.md

```diff
@@ -0,0 +1,140 @@
+---
+uid: github_reusable_workflows_nuget_solution_master_workflow
+---
+
+# NuGet solution master workflow
+
+> [!IMPORTANT]
+> This workflow is intended for internal use at Skyline Communications as it uses Skyline-specific validation. However, you can use it as a baseline and adjust it for use elsewhere by replacing the validation step with your own.
+
+The NuGet solution master workflow is designed to run on repositories containing the [DataMiner NuGet Package Solution](xref:Automation_scripts_as_a_Visual_Studio_solution) provided by the DIS extension in Visual Studio or from [Skyline.DataMiner.VisualStudioTemplates](https://www.nuget.org/packages/Skyline.DataMiner.VisualStudioTemplates#readme-body-tab).
+
+This workflow is a migration of the original [internal Jenkins pipelines](xref:Pipeline_stages_for_custom_solutions) used for automation and quality assurance within Skyline Communications.
+
+To use this workflow, your Visual Studio solution must consist of SDK-style projects. If you are migrating an existing library, you will need to convert it to SDK style. You can for example do so using the following dotnet tool: [Project2015To2017.Migrate2019.Tool](https://www.nuget.org/packages/Project2015To2017.Migrate2019.Tool). Alternatively, you can create a new solution with the template and then move all content from the old solution.
+
+The goal of this workflow is to automatically create and upload reusable .NET libraries if they pass certain quality standards. The workflow acts as a quality gate and code coverage collection before attempting to create and publish a NuGet package to nuget.org. Publishing only occurs during release cycles.
+
+- Skyline quality gate:
+
+  - [Validate NuGet metadata](#validate-nuget-metadata)
+  - [Building](#building)
+  - [Unit tests](#unit-tests)
+  - [Analyze](#analyze)
+  - [Quality gate](#quality-gate)
+
+  If the library passes these checks, the job will archive and provide the created NuGet package as an artifact, which the next job will attempt to sign.
```

Comment:
What is it that is archived: the job or the NuGet package? Could you rephrase this sentence so no confusion is possible?

Reply:
The NuGet package is archived, which means it becomes available as an Artifact in the workflow.
Perhaps:

If the library passes these checks, the job will archive the created NuGet package. This package is then available as an artifact on the workflow, which a user or the next jobs can access. The next job will attempt to sign this package.